server/src/main/java/org/apache/druid/server/coordinator/SegmentReplicantLookup.java

```diff
@@ -79,6 +80,7 @@ public static SegmentReplicantLookup make(DruidCluster cluster, boolean replicat
 
   private final Table<SegmentId, String, Integer> segmentsInCluster;
   private final Table<SegmentId, String, Integer> loadingSegments;
+  private final Map<SegmentId, Integer> segmentIdToReplicationFactor = new HashMap<>();
```

Comment:
Who clears this Map ? Is it generated every coordinator duty run. 

if yes then we should think of having an `int` as the value for performance reasons.

Reply:
Done